### PR TITLE
[quant][fx] Add pass in convert to fold quant-dequant sequence

### PR DESCRIPTION
**Pull request by supriyar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#54860 [quant][fx] Add pass in convert to fold quant-dequant sequence**
* #54859 [quant][fx][pyper] Get first linear use of quantize_per_tensor for FQN

Summary:
Currently we insert a quantize_per_tensor op when we encounter the quantizable input,
so if it has multiple uses and not all are quantizable then we need to add a dequantize op
before these ops.

In this pass - For a sequence of quantize_per_tensor - dequantize, we combine them
since it is a no-op.

Test Plan:
python test/test_quantization.py test_fold_quant_dequant

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27390506](https://our.internmc.facebook.com/intern/diff/D27390506)

### CODE REVIEW

File: torch/quantization/fx/quantize.py
Comment (cuspymd) [inline torch/quantization/fx/quantize.py:1007]: How about moving out of the for loop of `quant_uses`?

File: torch/quantization/fx/quantize.py
Comment (vkuzo) [inline torch/quantization/fx/quantize.py:1011]: would it be simple to inline it?

```
is_dequant = node.op == 'call_method' and node.target == "dequantize"
if is_dequant:
    ... do stuff ...
```

Asking because this file has a history of functions which are local to parent functions, and in those cases it's sometimes unclear which variables are passed in via args and which are from the parent function's scope.  If we need this to be a func, maybe we can move it to be a standalone function to make it easy to understand that all variables are passed in?

**Review by vkuzo — APPROVED** [review]
lgtm, perhaps would be good for @jerryzh168  to take a look as well?

File: torch/quantization/fx/quantize.py
Comment (supriyar) [reply torch/quantization/fx/quantize.py:1011]: makes sense to inline it. I thought there would be more to this function when I first wrote it.

**Review by jerryzh168 — APPROVED** [review]
Accepting to unblock pyper, we can revert the change after a proper fix is landed

File: test/quantization/test_quantize_fx.py
Comment (jerryzh168) [inline test/quantization/test_quantize_fx.py:2088]: remove?

File: test/quantization/test_quantize_fx.py
Comment (jerryzh168) [inline test/quantization/test_quantize_fx.py:2098]: we can also use self.checkGraphModuleNodes here I think, please take a look at other tests to see how to use it

**Pull request closed without merging**
